experiment-runner/ConfigValidator/Config/Validation/ConfigValidator.py

```diff
@@ -18,7 +21,47 @@ def __check_expression(name, value, expected, expression):
                 .config_values_or_exception_dict[name] = str(ConfigValidator.config_values_or_exception_dict[name]) + \
                                                     f"\n\n{ConfigAttributeInvalidError(name, value, expected)}"
             ConfigValidator.error_found = True
+    
+    # Verifies that an energybridge executable is present, and can be executed without error
+    @staticmethod
+    def __validate_energibridge(measure_enabled, eb_path):
+        # Do nothing if its not enabled
+        if not measure_enabled:
+            return
+
+        if  not platform.system() == "Linux"    \
+            or not os.path.exists(eb_path)      \
+            or not os.access(eb_path, os.X_OK):
+
+            ConfigValidator.error_found = True
+            ConfigValidator \
+            .config_values_or_exception_dict["EnergiBridge"] = "EnergiBridge executable was not present or valid"
+
+        # Test run to see if energibridge works
+        try:
+            eb_args = [eb_path, "--summary", "-o", "/dev/null", "--", "sleep", "0.5"]
+            p = subprocess.Popen(eb_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
```

Comment:
Might need `sudo` privileges

Reply:
So sudo privileges are not needed if you set EnergiBridge up correctly. They instruct you to add the binary to a group that gets access to the features it needs. That being intel RAPL. A model-specific register that allows energy metrics to be polled.

Reply:
Yep yep, you're right